### ISSUE TITLE
Requesting support for decaying the weight decay constant in weight_decay_optimizers.

### ISSUE THREAD

**Issue by sfleisch** [tensorflow/addons#865]
API type: Optimizer
Who will benefit: Anyone using a weight decay optimizer who wants to use learning rate decay.

The comments for the weight decay optimizers state that if learning rate decay is employed the weight decay scaling parameter should also be decayed with the same function (albeit most probably different initial and maximal values of the decay parameter.  With tfa 0.7.0, learning rate schedule  is supported, which it wasn't in 0.6.0, but it isn't for the weight decay parameter. I think that there needs to be a test for the weight decay being an instance of learning rate schedule class and executing the __call__ function with the global step when that is the case.

**Comment by WindQAQ** [comment]
cc @PhilJd
